### PR TITLE
Implemented schema name handling

### PR DESCRIPTION
**Pull request by khotilov**
I wasn't able to try out vertica.dplyr for awhile since it couldn't really handle schema names in any satisfactory fashion. I've spent some time this weekend to fix that. I hope you would find the approach rather transparent. Things seem to work for me so far, but I tried only limited number of operations.
- A general approach for handling schema names is implemented by using a fully specified table names as in '"MYSCHEMA"."MYTABLE"'. Default schema name could be set via the 'dplyr.vertica_default_schema' option, e.g., options(dplyr.vertica_default_schema='MYSCHEMA'). Initially, dplyr.vertica_default_schema points to 'public' schema. Tables in the default schema don't require full specification, and also they are listed without the schema name when printing a src_vertica object or when querying db_list_tables.
- Speed up the fairly frequently used db_has_table method a bit by querying only for a specific table instead of the whole list (which could be large).
- Make db_query_rows and db_query_fields to handle generic sql argument, and get rid of db_get_cols.
- Fixed code spacing a bit.

### CODE REVIEW

File: R/src-vertica.R [hunk 1]
```diff
@@ -398,14 +401,24 @@ db_list_tables.src_vertica <- function(src) {
...
 #' @export
 db_has_table.VerticaConnection <- function(con, table) {
-  res <- db_list_tables(con)
-  table %in% res
+  assert_that(is.string(table))
+
+  st <- get_schema_table(table)
+  tbl_query <- paste0("SELECT COUNT(*) N FROM all_tables WHERE lower(table_name)=\'", tolower(st$table),
+                      "\' AND lower(schema_name)=\'", tolower(st$schema),"\'")
+  if(con@type=="ODBC") {
```
Comment (etduwx) [inline R/src-vertica.R]: Instead of the if-clause, you can also use send_query here.

**Comment by etduwx** [comment]
@khotilov ,

Thanks very much for this pull request; they include some of the improvements that I had planned to make. The overall approach looks good. But so far there is one issue, one of my tests that used to pass now fails:

vertica_odbc <- src_vertica("VerticaDSN")

d1 <- data_frame(
  x = c(1,2,3),
  y = c(4,5,6),
  a = c("X","Y","Z")
  )

d2 <- data_frame(
  x2 = c(2,2,3),
  y2 = c(4,1,2),
  b = c("B","B","C")
  )

dd1 <- copy_to(vertica_odbc,d1,"d1",fast.load=FALSE)
dd2 <- copy_to(vertica_odbc,d2,"d2",fast.load=FALSE)
left_join(dd1, dd2, by = c("y" = "y2"))
Error: select is not a character vector

I'll also look into it further over the next couple of days. 

Thanks,
Edward

**Comment by khotilov** [comment]
The last db_query_fields commit should fix the issue.
And thanks for noticing the opportunity for send_query.

**Comment by etduwx** [comment]
Thanks, @khotilov . The test results are looking good on my end. I'll go ahead and make the merge.

We are grateful for your contribution.

**Pull request merged**